### PR TITLE
Fix metadata test

### PR DESCRIPTION
**Pull request by nathandunn**
- [x] fix delete organism by id: https://github.com/GMOD/Apollo/issues/2294
- [x] updating an organism in remote mode, publicMode is set to True while it should not: b81efae#diff-6ed0ac4ee19d8186384f9c6da0f89f1fR51: https://github.com/GMOD/Apollo/issues/2295
- [x] when updating organism, allow inhibition of data: https://github.com/GMOD/Apollo/issues/2293

### CODE REVIEW

**Comment by nathandunn** [comment]
Should be fixed with the Apollo `develop` branch now. 

@abretaud  You have to submit the JSON as strings as string, not JSON is native to the Apollo back-end.

File: README.rst [hunk 1]
```diff
@@ -1,3 +1,6 @@
+
+No longer synced.  `Actively maintained version <https://github.com/galaxy-genome-annotation/python-apollo>`__
```
Comment (abretaud) [inline README.rst]: Hum, not sure what you mean here?

File: bootstrap_apollo.sh [hunk 2]
```diff
@@ -3,7 +3,7 @@
 export GALAXY_SHARED_DIR=`pwd`/apollo_shared_dir
 mkdir -p "$GALAXY_SHARED_DIR"
 
-docker run -d -it -p 8888:8080 -v `pwd`/apollo_shared_dir/:`pwd`/apollo_shared_dir/ quay.io/gmod/apollo:2.4.1
+docker run --memory=4g -d -it -p 8888:8080 -v `pwd`/apollo_shared_dir/:`pwd`/apollo_shared_dir/ quay.io/gmod/apollo:latest
```
Comment (abretaud) [inline bootstrap_apollo.sh:6]: I don't think the --memory arg is needed, by default it's unlimited

**Review by abretaud — COMMENTED** [review]
Thanks! I've implemented the metadata another way in 25c61946faffb6c560680cb181813c41ced20552, to accept json or str
There's a test failing now that we use the "latest" image: https://github.com/galaxy-genome-annotation/python-apollo/commit/2b33b158f4cb1875d3af4a5b972f62a889847428#diff-deec9f621dff0d6025ee28ce963fc134R108 = deleting an organism by its commonname fails with error "problem deleting organism: java.lang.NumberFormatException: For input string: "temp_org"", do you see where it comes from?

**Comment by abretaud** [comment]
@nathandunn added another test failing: https://github.com/galaxy-genome-annotation/python-apollo/pull/17/commits/c82ad2d728dd85d0b085aa1a3dcae7ecfc6c5b74 = deleting an organism by its id with the remote api

**Comment by abretaud** [comment]
Another failing test: for some reason when updating an organism in remote mode, publicMode is set to True while it should not: https://github.com/galaxy-genome-annotation/python-apollo/pull/17/commits/b81efaed55a10ca199198c27c4c1b11f3122a857#diff-6ed0ac4ee19d8186384f9c6da0f89f1fR51

**Comment by abretaud** [comment]
[Another test case](https://github.com/galaxy-genome-annotation/python-apollo/pull/17/commits/5835cb3b206cf6a9b75fa5e9f82a551dd9a17547) that fails: when updating an organism with the same fasta but a different directory, sequences are duplicated in the db

(sorry for the flow of bug reports!)

File: bootstrap_apollo.sh (hunk 2 above)
Comment (nathandunn) [reply bootstrap_apollo.sh:6]: https://docs.docker.com/v17.12/docker-for-mac/

Its 2 GB by default on the mac, so it is super-sluggish unless I set this.

File: README.rst (hunk 1 above)
Comment (nathandunn) [reply README.rst]: That was from my old repo.  Should be gone now.

Comment (nathandunn) [reply README.rst]: I think your changes are better.  I merged those in.

**Comment by nathandunn** [comment]
> [Another test case](https://github.com/galaxy-genome-annotation/python-apollo/pull/17/commits/5835cb3b206cf6a9b75fa5e9f82a551dd9a17547) that fails: when updating an organism with the same fasta but a different directory, sequences are duplicated in the db 

That is the correct implementation.   It duplicates the sequence in the database, since a sequence is owned by an organism.

> (sorry for the flow of bug reports!)

Not a problem.  I've added them to fix up at the top.

**Comment by abretaud** [comment]
> That is the correct implementation. It duplicates the sequence in the database, since a sequence is owned by an organism.

Arf I didn't know it was implemented like this, did it change or was it always like this? So the annotations are lost?
I often want to update an org like this just because I generated a new jbrowse in galaxy with the same reference sequence, but different tracks/display settings
Would it be possible (optional?) to reuse the old sequence in the db if they are the same in the new data dir (same name+length)?

**Comment by nathandunn** [comment]
> Arf I didn't know it was implemented like this, did it change or was it always like this? So the annotations are lost?
> I often want to update an org like this just because I generated a new jbrowse in galaxy with the same reference sequence, but different tracks/display settings
> Would it be possible (optional?) to reuse the old sequence in the db if they are the same in the new data dir (same name+length)?

I think I was talking about duplicating genomes.    I'll add it here: https://github.com/GMOD/Apollo/issues/2293

**Comment by nathandunn** [comment]
Incorrectly closed this when merging a separate repository (!!!)

**Comment by nathandunn** [comment]
@abretaud  Are you planning on using the travis cache?

**Comment by nathandunn** [comment]
All tests with local apollo, but still getting failures with the docker versions.

**Comment by nathandunn** [comment]
@abretaud  this works for me locally:  

image

```
__default: local
local:
    url: "http://localhost:8080/apollo"
    username: "admin@local.host"
    password: "password"
```

**Comment by nathandunn** [comment]
@abretaud  Any ideas why we get the 4 errors from the quay build?

**Comment by abretaud** [comment]
@nathandunn I think the 2 org deletion errors come from the change in the api at https://github.com/GMOD/Apollo/pull/2298/files#diff-dd23bc0329e522c1d6a07295b706dc1fR53 -> would it be possible to keep the same param name as before? I think it's never desirable to change an api signature, and here I think "organism" is as good as "id"
I'll look at the 2 others

**Comment by nathandunn** [comment]
> @nathandunn I think the 2 org deletion errors come from the change in the api at https://github.com/GMOD/Apollo/pull/2298/files#diff-dd23bc0329e522c1d6a07295b706dc1fR53 -> would it be possible to keep the same param name as before? I think it's never desirable to change an api signature, and here I think "organism" is as good as "id"
> I'll look at the 2 others

I should have backported the changes so it will work with either.    


I think what is going on (at least locally for me) is that quay.io not pulling the latest `latest` image.  Even locally its given me an image that is 10 days old.    That explains why things were breaking with docker, though working locally. 

I'll update the API to provide for both in a bit.

**Comment by nathandunn** [comment]
@abretaud  / @erasche   

So good news / bad news.  

1. works against a local Apollo just fine, all tests pass
1. works against a local DOCKER build just fine (`docker build .`), all tests pass
1. fails (4) against an automated docker build against quay.io or docker hub builds of either develop or the tag
1. works, build locally and pushing up tag (`quay.io/gmod/apollo:local-build-test`), all tests pass. 



I think the problem is that it is pulling an older version (instead of the current has):

https://github.com/GMOD/Apollo/blob/develop/Dockerfile#L24

Created this issue to fix this: https://github.com/GMOD/Apollo/issues/2299

I'm going to re-run this against latest.  Not sure why the `develop` doesn't update on the github site, but its always going to be a version behind regardless.

**Comment by nathandunn** [comment]
@abretaud  So all of the code fixes we did worked, but the docker image being built remotely wasn't getting the newest image.  I updated it to use `git clone ` directly (it always pulls off of `develop` at the moment), but if you have a better idea of a way to do this.     

This test should work regardless now.

**Comment by nathandunn** [comment]
@abretaud  I think that this can / should be merged (pending review). 

Let me know if there should be further changes.

**Pull request merged**